Add UUID support

armin-salihovic opened the pull request:
This feature introduces support for UUIDs in the `nestedSetWithUuid` macro:

```php
Schema::create('table', function (Blueprint $table) {
    ...
    $table->nestedSetWithUuid();
});
```

This feature is tested under the same test cases as the auto-increment IDs, ensuring consistency across both ID types.

armin-salihovic committed:
Add UUID support

armin-salihovic committed:
Rename parameter uuid to withUuid

aimeos commented:
@lazychaser @jonnott
Any plans to merge this PR and add UUID support?

aimeos commented:
> @aimeos This project is essentially abandoned, so I don't think there's any plan to add new features.
> 
> I've been made a maintainer, but all I'm going to be doing is keeping it alive and useful (as-is), as we move to newer PHP/Laravel versions.

@jonnott We can offer to co-maintain the package to ensure further development. We are already maintaining a lot of packages, mostly our own but also others: https://github.com/aimeos?tab=repositories

aimeos commented:
@jonnott As you've said that the package is practically abandoned, we've created a fork, merged all open PRs that make sense, fixed some issues and added full support for SQL Server. The new package (aimeos/laravel-nestedset) is now available here: https://laravel-nestedset.org

The new package is compatible to this one but uses a different namespace for the classes/trait.

If you don't want to support this package any more, it would be great if you would update the README to state that and point to aimeos/laravel-nestedset as alternative.

armin-salihovic commented:
> @jonnott As you've said that the package is practically abandoned, we've created a fork, merged all open PRs that make sense, fixed some issues and added full support for SQL Server. The new package (aimeos/laravel-nestedset) is now available here: https://laravel-nestedset.org
> 
> The new package is compatible to this one but uses a different namespace for the classes/trait.
> 
> If you don't want to support this package any more, it would be great if you would update the README to state that and point to aimeos/laravel-nestedset as alternative.

Amazing news! 
Happy to see the code I wrote did not go to waste.

Thank you for your effort! A star from me 👍

aimeos commented:
Thanks! This PR has been merged into https://github.com/aimeos/laravel-nestedset

aimeos commented:
@jonnott Noticed that the old tags have been pushed too, which was by accident.
Due to new features (additional depth column) and the new namespace, the aimeos/laravel-nestedset package isn't a drop-in replacement, more a 7.x version and you need to care about two things described in the migration section: https://laravel-nestedset.org/#migration

If you only changed kalnoy/laravel-nestedset to aimeos/laravel-nestedset in your composer.json, nothing has happened because you are still using the old code base.

aimeos commented:
> Why not make it v7.0 then?

This is a valid question and was mainly due to the namespace change.
To make updating easier, the v6.0.6 tag has been re-added and the new code base is now available as 7.0+

aimeos commented:
> Out of interest, what's behind the need for the new 'depth' column?

Better query performance for depth and SQL Server didn't support the query generated by withDepth().
